games/naval_battle.c

```diff
@@ -0,0 +1,965 @@
+/**
+ * @file
+ * @author [Carlos Rafael](https://github.com/CarlosZoft)
+ * @author [Herick Lima](https://github.com/hericklima22)
+ * @brief [naval_battle](https://en.wikipedia.org/wiki/Battleship_(game))
+ * implementation in C using only the stdio.h for Standard Input and Output.
+ * @details  Naval battle is a game, to be played by two people. It consists of
+ * knocking down the enemy ship, through shots , when hit the ship is
+ * revealed with the respective number of its size. Example: size 3 = 3 3 3 on
+ * the board.
+ * To play - boats over size 1, need direction; V -> vertical and H ->
+ * horizontal. Example Input 1 A H -> line 1, column A, direction H
+ * (Horizontal).
+ */
+
+#include <stdio.h>  /// for Standard Input Output
+
+/**
+ * @brief Function validEntryLineColumn
+ * Responsible for validating entries, for positioning boats
+ * @param line matrix row
+ * @param column matrix column
+ * @returns if the row and column are valid
+ */
+int validEntryLineColumn(int line, char column)
+{
+    if ((line >= 1 && line <= 10) && (column >= 65 && column <= 74))
+    {
+        return 1;
+    }
+
+    else
+    {
+        return 0;
+    }
+}
+/**
+ * @brief Function validatePosition
+ * Responsible for checking if the position can receive the boat.
+ * @param mat board
+ * @param boat boat
+ * @param line matrix row
+ * @param column matrix column
+ * @returns if the position is valid
+ */
+int validatePosition(int mat[10][10], int boat, int line, int column,
+                     char guide)
+{
+    int cont = 0;
+    int i, j;
+
+    if (boat < 1 || boat > 3)
+    {
+        return 0;
+    }
+
+    if (guide != 'H' && guide != 'V')
+    {
+        return 0;
+    }
+
+    if ((line < 0 || line > 9) || (column < 0 || column > 9))
+    {
+        return 0;
+    }
```

Comment:
under all these conditions function is returning `0`, therefore instead of writitng them in different blocks you can merge them into a single conditional block